### PR TITLE
Feature/paged response sublists

### PR DESCRIPTION
**Pull request by ffleandro**
Added auxiliary constructors for trimming paged lists;
Added more base Exceptions for handling generic errors;

Merges branch feature-entando-arch that brings some functionalities from entando-core for filtering and paginating responses.

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: src/main/java/org/entando/web/controller/ControllerExceptionHandler.java [hunk 1]
```diff
@@ -52,7 +52,8 @@ public ControllerExceptionHandler(final MessageSource messageSource) {
 
 	@ExceptionHandler(IllegalArgumentException.class)
 	public ResponseEntity<ErrorResponse> handleException(final IllegalArgumentException exception) {
-		return ResponseEntity.status(400).body(new ErrorResponse(exception.getMessage()));
+		final String message = messageSource.getMessage(exception.getMessage(), null, Locale.getDefault());
+		return ResponseEntity.status(400).body(new ErrorResponse(message));
```
Comment (werbth) [inline src/main/java/org/entando/web/controller/ControllerExceptionHandler.java]: Use the constant HttpStatus.BAD_REQUEST for status method.

**Review by werbth — CHANGES_REQUESTED** [review]

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Review by werbth — APPROVED** [review]

**Pull request merged**